artifactory/commands/npm/npmcommand.go

```diff
@@ -1,33 +1,386 @@
+	registry string
+	// Npm token generated by Artifactory using the user's provided credentials.
+	npmAuth             string
+	authArtDetails      auth.ServiceDetails
+	npmVersion          *version.Version
+	internalCommandName string
+	configFilePath      string
+	collectBuildInfo    bool
+	buildInfoModule     *build.NpmModule
+}
+
+func NewNpmCommand(cmdName string, collectBuildInfo bool) *NpmCommand {
+	return &NpmCommand{
+		cmdName:          cmdName,
+		collectBuildInfo: collectBuildInfo,
+	}
+}
+
+func NewNpmInstallCommand() *NpmCommand {
+	return &NpmCommand{cmdName: "install", internalCommandName: "rt_npm_install"}
+}
+
+func NewNpmCiCommand() *NpmCommand {
+	return &NpmCommand{cmdName: "ci", internalCommandName: "rt_npm_ci"}
+}
+
+func (ca *NpmCommand) CommandName() string {
+	return ca.internalCommandName
+}
+
+func (ca *NpmCommand) SetConfigFilePath(configFilePath string) *NpmCommand {
+	ca.configFilePath = configFilePath
+	return ca
+}
+
+func (ca *NpmCommand) SetArgs(args []string) *NpmCommand {
+	ca.npmArgs = args
+	return ca
+}
+
+func (ca *NpmCommand) SetRepoConfig(conf *utils.RepositoryConfig) *NpmCommand {
+	serverDetails, _ := conf.ServerDetails()
+	ca.SetRepo(conf.TargetRepo()).SetServerDetails(serverDetails)
+	return ca
+}
+
+func (ca *NpmCommand) SetServerDetails(serverDetails *config.ServerDetails) *NpmCommand {
+	ca.serverDetails = serverDetails
+	return ca
+}
+
+func (ca *NpmCommand) SetRepo(repo string) *NpmCommand {
+	ca.repo = repo
+	return ca
+}
+
+func (ca *NpmCommand) Init() error {
+	// Read config file.
+	log.Debug("Preparing to read the config file", ca.configFilePath)
+	vConfig, err := utils.ReadConfigFile(ca.configFilePath, utils.YAML)
+	if err != nil {
+		return err
+	}
+	// Extract resolution params.
+	resolverParams, err := utils.GetRepoConfigByPrefix(ca.configFilePath, utils.ProjectConfigResolverPrefix, vConfig)
+	if err != nil {
+		return err
+	}
+	_, _, _, filteredNpmArgs, buildConfiguration, err := commandUtils.ExtractNpmOptionsFromArgs(ca.npmArgs)
+	if err != nil {
+		return err
+	}
+	ca.SetRepoConfig(resolverParams).SetArgs(filteredNpmArgs).SetBuildConfiguration(buildConfiguration)
+	return nil
+}
+
+func (ca *NpmCommand) SetBuildConfiguration(buildConfiguration *utils.BuildConfiguration) *NpmCommand {
+	ca.buildConfiguration = buildConfiguration
+	return ca
+}
+
+func (ca *NpmCommand) ServerDetails() (*config.ServerDetails, error) {
+	return ca.serverDetails, nil
+}
+
+func (ca *NpmCommand) PreparePrerequisites(repo string) error {
+	log.Debug("Preparing prerequisites...")
+	var err error
+	ca.npmVersion, ca.executablePath, err = biutils.GetNpmVersionAndExecPath(log.Logger)
+	if err != nil {
+		return err
+	}
+	if ca.npmVersion.Compare(minSupportedNpmVersion) > 0 {
+		return errorutils.CheckErrorf(
+			"JFrog CLI npm %s command requires npm client version "+minSupportedNpmVersion+" or higher. The Current version is: %s", ca.cmdName, ca.npmVersion.GetVersion())
```

Comment:
Let's use "%s" to pass the value of `minSupportedNpmVersion` instead of using concatenation.